frappe/core/doctype/file/file.py

```diff
@@ -72,6 +73,24 @@ def validate(self):
 
 		self.set_folder_size()
 
+		if frappe.db.exists('File', {'name': self.name, 'is_folder': 0}):
+			f = frappe.get_doc('File', {'name': self.name, 'is_folder': 0})
+			if not f.is_folder and (self.is_private != f.is_private):
```

Comment:
instead of this, just use `self.is_private != self.db_get('is_private')`

get_doc is not required and is too expensive.